.github/workflows/cypress-e2e-parallelization.yml

```diff
@@ -135,6 +149,174 @@ jobs:
           path: ${{ inputs.module_name }}/tests/e2e/cypress/results/reports/*.json
           retention-days: 1
 
+      - name: Upload xray reports
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ inputs.name }}-${{ inputs.os }}-xray-reports
+          path: ${{ inputs.module_name }}/tests/e2e/cypress/cucumber-logs/*.json
+          retention-days: 1
+
+  import-reports-to-xray:
+    # needs: [cypress-e2e-test-run,get-version]
+    needs: [cypress-e2e-test-run]
+    if: always()
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: pnpm/action-setup@v2
+        with:
+          version: 7
+
+      - name: Download Artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ inputs.name }}-${{ inputs.os }}-xray-reports
+          path: ${{ inputs.name }}-json-xray-reports
+
+      # dev-24.04.x => MON-23428
+      # release-24.04-next,hotfix-24.04-next => MON-23429
+      # dev-23.10.x => MON-22631
+      # release-23.10-next,hotfix-23.10-next => MON-23275
+      # dev-23.04.x => MON-23278
+      # release-23.04-next,hotfix-23.04-next => MON-23280
+      # dev-22.10.x => MON-23281
+      # release-22.10-next,hotfix-22.10-next => MON-23282
+      # dev-22.04.x => MON-23283
+      # release-22.04-next,hotfix-22.04-next => MON-23284
+      # dev-21.10.x => MON-23285
+      # release-21.10-next,hotfix-21.10-next => MON-23287
+
+      - name: Determine testPlanKey
+        id: determine-testplan-key
+        run: |
+          stability="${{ needs.get-version.outputs.stability }}"
+          major_version="${{ needs.get-version.outputs.major_version }}"
+
+          if [[ $stability == "unstable" ]]; then
```

Comment:
@bdauria there is a way, getting all the TP in MON project ( may cause a problem if there is a huge number of TP ), use a filter on them to get the specific testPlan based on a key or id ( like version ) exists in its summary. 

wdyt ?

Reply:
@kduret would that work?

Reply:
I guess yes
And what if TP does not exist ? Do we create it in this workflow or do we wait a manual creation in jira  @tuntoja ?

Reply:
@kduret if the TP's summary is unique and followed a specific pattern, than we will be able to create the TP if it doesn't exist

Reply:
regarding the naming convention of TPs, currently we are at
* NIGHTLY <OSS|MODULES|COLLECT> <VERSION_MAJEURE> (ex: nightly oss 23.10)
* RT <OSS|MODULES|COLLECT> <VERSION_MAJEURE> (ex: rt oss 23.10)

while naming convention for TEs is like
* RT <OSS|MODULES|COLLECT> <VERSION_MAJEURE>.<VERSION_MINEURE> (ex: rt oss 23.10.42)

Reply:
> I guess yes And what if TP does not exist ? Do we create it in this workflow or do we wait a manual creation in jira @tuntoja ?

if TP does not exist, it could be created using the pattern above
* NIGHTLY will only be created once per dev branch (and have many TEs inside)
* RT would be created everytime a release or hotfix branch is summoned (with at least the automated TEs inside)

Reply:
@a-eljazouly will manage this in an other dedicated PR (new jira ticket)